dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/ValidationStatusCountMetric.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.core.metrics;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dockstore.webservice.core.metrics.ValidationExecution.ValidatorTool;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.util.HashMap;
+import java.util.Map;
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.MapKeyEnumerated;
+import javax.persistence.Table;
+import org.hibernate.annotations.BatchSize;
+
+@Entity
+@Table(name = "validation_status")
+@ApiModel(value = "ValidationStatusMetric", description = "Aggregated metrics about workflow validation statuses")
+@Schema(name = "ValidationStatusMetric", description = "Aggregated metrics about workflow validation statuses")
+@SuppressWarnings("checkstyle:magicnumber")
+public class ValidationStatusCountMetric extends CountMetric<ValidatorTool, Boolean> {
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    @MapKeyColumn(name = "validatortool")
+    @MapKeyEnumerated(EnumType.STRING)
+    @Column(name = "isvalid", nullable = false)
+    @CollectionTable(name = "validator_tool_validation_status", joinColumns = @JoinColumn(name = "validationstatusid", referencedColumnName = "id"))
+    @BatchSize(size = 25)
+    @ApiModelProperty(value = "A map containing key-value pairs indicating whether the validator tool successfully validated the workflow", required = true)
+    @Schema(description = "A map containing key-value pairs indicating whether the validator tool successfully validated the workflow", required = true, example = """
+            {
+                "MINIWDL": true
+            }
+            """)
+    private Map<ValidatorTool, Boolean> validatorToolToIsValid = new HashMap<>();
```

Comment:
A bit confused because this is an aggregated metric with a map from validator tool to a single boolean. 
What does this return if a validator tool is changing (being upgraded) over time and thus validates success or false over time?

Reply:
The boolean here represents the aggregated `isValid` value from all validations, which is calculated by the metrics aggregator.

>  What does this return if a validator tool is changing (being upgraded) over time and thus validates success or false over time? 

> But what are we expecting here? The latest validation value?

Currently, the metrics aggregator performs an `&&` across all validation booleans and considers that the aggregated value. It doesn't take into account success/failure over time... if it did/if we want it to, the metrics aggregator should probably take the latest value.

Did you have a different idea about how the aggregated metric should look like?

Reply:
I think this thread overlaps with my other comments. It does seem odd that once it's failed, it can never be valid again.

See Denis' [other comment](https://github.com/dockstore/dockstore/pull/5447#issuecomment-1490666310)... If we go with the "overwrite" as suggested there, doesn't that mean this shouldn't be an aggregate?